server/src/views/profile.js

```diff
@@ -0,0 +1,249 @@
+/* jshint browser:true */
+
+const React = require("react");
+const errors = require("../../shared/errors");
+
+function delay(duration) {
+  return new Promise(resolve => setTimeout(resolve, duration));
+}
+
+exports.ProfileButton = class ProfileButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onDocumentClick = this.dismiss.bind(this);
+    this.onRefreshProfileBound = this.onRefreshProfile.bind(this);
+    this.state = {
```

Comment:
I could make them props, and have `Frame` pass a function that updates the profile. In that case, though, wouldn't `avatar`, `nickname`, and `email` still need to be state (just on `Frame`, not `ProfileButton`)?